src/redux/actions/rewards.js

```diff
@@ -71,26 +71,31 @@ export function doClaimRewardType(rewardType, options = {}) {
     });
 
     const success = successReward => {
-      dispatch({
-        type: ACTIONS.CLAIM_REWARD_SUCCESS,
-        data: {
-          reward: successReward,
-        },
-      });
-      if (
-        successReward.reward_type === rewards.TYPE_NEW_USER &&
-        rewards.callbacks.claimFirstRewardSuccess
-      ) {
-        rewards.callbacks.claimFirstRewardSuccess();
-      } else if (successReward.reward_type === rewards.TYPE_REFERRAL) {
-        dispatch(doFetchInviteStatus());
-      }
-
-      dispatch(doRewardList());
-
-      if (options.callback) {
-        options.callback();
-      }
+      // Temporary timeout to ensure the sdk has the correct balance after claiming a reward
+      setTimeout(() => {
+        dispatch(doUpdateBalance()).then(() => {
+          dispatch({
+            type: ACTIONS.CLAIM_REWARD_SUCCESS,
+            data: {
+              reward: successReward,
+            },
+          });
+          if (
+            successReward.reward_type === rewards.TYPE_NEW_USER &&
+            rewards.callbacks.claimFirstRewardSuccess
+          ) {
+            rewards.callbacks.claimFirstRewardSuccess();
+          } else if (successReward.reward_type === rewards.TYPE_REFERRAL) {
+            dispatch(doFetchInviteStatus());
+          }
+
+          dispatch(doRewardList());
+
+          if (options.callback) {
+            options.callback();
+          }
+        });
+      }, 1000);
```

Comment:
While I understand the reasoning behind this,I feel it's a bit weird because it may be possible for the sdk to not have the updated balance after a second. Maybe we could also explore using messages like "your wallet balance will be updated shortly" or something similar. Not really a blocker though.

Reply:
Yeah, I agree it's not a very good solution. I think the better fix is to wait for the SDK to add balance updates through sockets.